File: javastory.budgetsh.stage4/src/javastory/budgetsh/stage4/server/budget/store/TransactionStore.java
package javastory.budgetsh.stage4.server.budget.store;

import java.util.Collection;

import javastory.budgetsh.stage4.server.budget.entity.budget.Transaction;

public interface TransactionStore {
	//
	void remove(String id);
	void update(Transaction transaction);
	boolean exist(String id);
	Transaction retrieve(String id, String id2);
	boolean register(Transaction transaction);
	Collection<Transaction> retrieveAll(String id);

}

File: javastory.budgetsh.stage5.1/src/step1/store/TransactionStore.js
const MapStorage = require('./MapStorage.js');
const mapStorage = new MapStorage();

class TransactionStore{
	constructor(){
		
	}
	exist(id){
		return mapStorage.transactionMap.has(id);
	}
	
	register(transaction){
		mapStorage.transactionMap.set(transaction.id, transaction);
	}
	
	find(id){
		return mapStorage.transactionMap.get(id);
	}
	
	remove(id){
		mapStorage.transactionMap.delete(id);
	}
}

module.exports = TransactionStore;